### PR TITLE
Changed RTL processing of carousel control icons

### PR DESCRIPTION
**Pull request by 3baaady07**
### Description
Changed the way RTL processes carousel icons (`carousel-control-prev-icon` and `carousel-control-next-icon`) . The processing originally uses `rtl:options`, whereas now it uses an inline `rtl:{value}` to flip the values instead of changing the selector name.

### Motivation & Context

1. This was done to enable tools such as [`postcss-rtlcss`](https://github.com/elchininet/postcss-rtlcss) to process the files and generate a file with both `[dir=rtl]` and `[dir=ltr]` prefixed selectors.
2. It was also to prevent `rtl:options` from affecting css that is scoped outside of the intended `rtl:options` block.

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [ ] My change introduces changes to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

#### Live previews

- https://deploy-preview-39536--twbs-bootstrap.netlify.app/docs/5.3/components/carousel/
- https://deploy-preview-39536--twbs-bootstrap.netlify.app/docs/5.3/examples/cheatsheet-rtl/#carousel

### CODE REVIEW

File: scss/_carousel.scss
Comment (julien-deramond) [inline scss/_carousel.scss:136]: Here's a suggestion to avoid having extra spaces generated:

```suggestion
  background-image: escape-svg($carousel-control-prev-icon-bg) #{"/*rtl:" + escape-svg($carousel-control-next-icon-bg) + "*/"};
```

Here's a diff between your version and the suggested one:

```diff
- background-image:  url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23fff'%3e%3cpath d='M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z'/%3e%3c/svg%3e") ;
+ background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23fff'%3e%3cpath d='M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z'/%3e%3c/svg%3e");
```

File: scss/_carousel.scss
Comment (julien-deramond) [inline scss/_carousel.scss:139]: Suggestion in the same spirit for this line:

```suggestion
  background-image: escape-svg($carousel-control-next-icon-bg) #{"/*rtl:" + escape-svg($carousel-control-prev-icon-bg) + "*/"};
```

**Review by julien-deramond — COMMENTED** [review]
Thanks for this PR @3baaady07

I don't mind changing the approach if it is better supported by external tools and if the generated CSS is the same. My knowledge on the topic is rather limited so @ffoodd, if you have a little bit of time to double-check it too :) Maybe there was a specific reason I'm not aware of to use `rtl:options` 🤷 

On the generated aspect. I've compared both `bootstrap.rtl.css`. Here's the diff:

```diff
ju ߷ ~/t/bootstrap ➤ (main) diff dist/css/bootstrap.rtl.css dist/css/bootstrap.new.rtl.css                                                               (0.000002 hrs)
6126,6128d6125
< }
< .carousel-control-next-icon {
<   background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23fff'%3e%3cpath d='M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z'/%3e%3c/svg%3e");
6132a6130,6133
> }
> 
> .carousel-control-next-icon {
>   background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23fff'%3e%3cpath d='M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z'/%3e%3c/svg%3e");
6186,6187c6187,6188
< .carousel-dark .carousel-control-next-icon,
< .carousel-dark .carousel-control-prev-icon {
---
> .carousel-dark .carousel-control-prev-icon,
> .carousel-dark .carousel-control-next-icon {
6197,6199c6198,6200
< [data-bs-theme=dark] .carousel .carousel-control-next-icon,
< [data-bs-theme=dark] .carousel .carousel-control-prev-icon, [data-bs-theme=dark].carousel .carousel-control-next-icon,
< [data-bs-theme=dark].carousel .carousel-control-prev-icon {
---
> [data-bs-theme=dark] .carousel .carousel-control-prev-icon,
> [data-bs-theme=dark] .carousel .carousel-control-next-icon, [data-bs-theme=dark].carousel .carousel-control-prev-icon,
> [data-bs-theme=dark].carousel .carousel-control-next-icon {
```

The only difference relies on an extra break line (which is good) and the order of declarations: `.carousel-control-prev-icon` and `.carousel-control-next-icon` are now inverted. But it doesn't change the spirit or the final rendering since there's not especially a rule depending on the other one in terms of priority.

The rendering and user experience are the same as before (see https://deploy-preview-39536--twbs-bootstrap.netlify.app/docs/5.3/examples/cheatsheet-rtl/#carousel).

File: scss/_carousel.scss
Comment (3baaady07) [reply scss/_carousel.scss:139]: Its' definitely good to know that the `+` sign works as a concat operator. 😅

**Comment by 3baaady07** [comment]
Hi @julien-deramond,

What's the timeline for making a decision regarding this PR?

**Review by julien-deramond — APPROVED** [review]
I've updated the PR to take into account the comments so that it handles correctly the whitespaces. Based on my current knowledge of the topic (see https://github.com/twbs/bootstrap/pull/39536#pullrequestreview-1798746669), I think it's safe to merge.

> What's the timeline for making a decision regarding this PR?

@3baaady07 It can take a little bit of time because it'll depend on the availability of the other maintainers in their spare time to confirm that this approach doesn't imply any regressions. But there probably won't be any other actions on your end, either it will be approved and merged as is, or closed if there's an issue with this approach.

On my side, I'll be off for a couple of weeks. I'll check back on the status of this PR when I'll get back.

**Comment by 3baaady07** [comment]
Thank you @julien-deramond,
I'll use [`patch-package`](https://www.npmjs.com/package/patch-package) till a decision is made 😊
Cheers 🤙

**Review by louismaximepiton — COMMENTED** [review]
Seems easier to maintain indeed, the only drawback I can see here is that the `bootstrap.css` increases its size by `.04kB`. So I'd be in favor of this change if it can help people.
Another solution to scope the changes could be to use `/* rtl:begin:options */` and `/* rtl:end:options */` [source](https://rtlcss.com/learn/usage-guide/string-map/)

**Comment by 3baaady07** [comment]
Hi @louismaximepiton,

The trade-off to the increase in size is to enable [`postcss-rtlcss`](https://github.com/elchininet/postcss-rtlcss) to process the css and generate a single, optimized file with selectors prefixed with `[dir=rtl]` and `[dir=ltr]`. `postcss-rtlcss` is a plugin promoted by the reputable module bundler plugin [`postcss`](https://github.com/postcss/postcss), however `postcss-rtlcss` does not process `/* rtl:options */` directive or any of its derivatives.

The changes proposed enables the plugin to process the css file while functionally maintaining the [`rtlcss`](https://rtlcss.com/) output.

Cheers,

**Review by ffoodd — APPROVED** [review]
LGTM :rocket: 

Thanks for the PR!

**Pull request merged**
